synapse/daemon.py

```diff
@@ -359,14 +387,109 @@ def _getTaskFiniMesg(self, task, valu):
         typename = valu.typename
         return ('task:fini', {'task': task, 'retn': retn, 'type': typename})
 
+    async def _onTaskV2Init(self, link, mesg):
+
+        # t2:init is used by the pool sockets on the client
+
+        name = mesg[1].get('name')
+        sidn = mesg[1].get('sess')
+        todo = mesg[1].get('todo')
+
+        try:
+
+            if sidn is None or todo is None:
+                raise s_exc.NoSuchObj(name=name)
+
+            sess = self.sessions.get(sidn)
+            if sess is None:
+                raise s_exc.NoSuchObj(name=name)
+
+            item = sess.getSessItem(name)
+            if item is None:
+                raise s_exc.NoSuchObj(name=name)
+
+            s_scope.set('sess', sess)
```

Comment:
Where is this used? I don't see any use of `s_scope.get('sess')` in this pr

Reply:
This is mostly so that a shared object being called ( this message represents a new asyncio Task ) can gain access to their telepath session if they need to.  I could probably remove it for now, but it'd likely go back in later :D